index.js

```diff
@@ -1,43 +1,62 @@
 /* eslint-env node */
 'use strict';
+const Funnel = require('broccoli-funnel');
+const getDebugMacros = require('./src/debug-macros').debugMacros;
 
-function debugMacros(features) {
-  let plugins = [
-    [
-      require.resolve('babel-plugin-debug-macros'),
-      {
-        flags: [
-          {
-            source: 'ember-m3/feature-flags',
-            flags: features,
-          },
-        ],
-      },
-      '@ember-data/canary-features-stripping',
-    ],
-  ];
-
-  return plugins;
+/*
+ shouldIncludeChildAddon runs during
+ addon init so does not have access to
+ the host application instance.
+*/
+function hasDataPackage(addon) {
+  let addons = addon.project.addonPackages;
```

Comment:
If possible using `project.findAddonByName` may be a more resilient approach as technically ember-cli has some addon initialization that occurs: https://github.com/ember-cli/ember-cli/blob/master/lib/models/project.js#L634

(I added this, specifically for this use-case)

Reply:
I investigated this and that util won't work for us because it does not recurse the addon dep tree, otherwise it would be very helpful.